[Docs] Add documentation and examples for `Figure` (#524)
mckinsey/vizro: vizro-core/docs/pages/user-guides/custom-charts.md

This guide shows you how to create custom charts and how to add them to your dashboard.
The [`Graph`][vizro.models.Graph] model accepts the `figure` argument, where you can enter _any_ [`plotly.express`](https://plotly.com/python/plotly-express/) chart as explained in the [user guide on graphs](graph.md).

### When to use a custom chart

> In general, the usage of the custom chart decorator `@capture("graph")` is required if your plotly chart requires any post-update calls or customization e.g:

Reviewer: ```suggestion
In general, you should use the custom chart decorator `@capture("graph")` if your plotly chart needs any post-update calls or customization e.g:
```

Reviewer: Is Vale running OK? I would have expected an error for the e.g. here. When you've successfully merged this PR, next sprint when I am back on Vizro tickets, I'll take a look to find out why it didn't.

Author: Interesting - yes, there are no errors raised or at least it's passing CI!
